packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx

```diff
@@ -28,32 +50,58 @@ import CardHeader from '../../../components/CardHeader';
 
 export const viewName = 'account-recovery-reset-password';
 
+const accountsEmail = 'accounts@firefox.com';
+
 export type AccountRecoveryResetPasswordProps = {
-  email: string;
-  linkStatus: LinkStatus;
-  forceAuth?: boolean;
-};
+  overrides?: {
+    navigate?: NavigateFn;
+    locationContext?: GenericContext;
+    urlSearchContext?: UrlSearchContext;
+  };
+} & RouteComponentProps;
 
 type FormData = {
   newPassword: string;
   confirmPassword: string;
 };
 
-type LinkStatus = 'expired' | 'damaged' | 'valid';
+enum BannerState {
+  None,
+  UnexpectedError,
+  PasswordResendSuccess,
+  PasswordResetSuccess,
+  Redirecting,
+  PasswordResendError,
+  InvalidContext,
+}
 
 const AccountRecoveryResetPassword = ({
-  email,
-  linkStatus,
-  forceAuth = false,
-}: AccountRecoveryResetPasswordProps & RouteComponentProps) => {
+  overrides,
+}: AccountRecoveryResetPasswordProps) => {
   usePageViewEvent(viewName, REACT_ENTRYPOINT);
 
+  const notifier = useNotifier();
+  const broker = useBroker();
+  const account = useAccount();
+  const relier = useRelier();
+
+  let navigate = useNavigate();
+  navigate = overrides?.navigate || navigate;
+
+  let urlSearchContext = useUrlSearchContext();
+  urlSearchContext = overrides?.urlSearchContext || urlSearchContext;
+
+  let locationContext = useLocationContext();
+  locationContext = overrides?.locationContext || locationContext;
+
+  const verificationInfo = new VerificationInfo(urlSearchContext);
+  const accountRecoveryKeyInfo = new AccountRecoveryKeyInfo(locationContext);
```

Comment:
Q: why/when do we need `urlSearchContext` vs `locationContext`? I see [this comment](https://github.com/mozilla/fxa/pull/14918/files#diff-11580915f1ec0366c245da9aff14961ad9c0d1c3595f8d6ff8bc88441de48ac5R16-R19), but still not sure when to use one or the other.

Reply:
UrlSearch is the query string. Location is the reach router location.state. In this case, some props are passed in the query string, but some are passed via the location state from the previous page. 

I guess they are just two different 'contexts', for lack of a better term. As far as deciding when to use the query string vs, the location state. I am not really sure... That's just what was done in the preceding step. I don't see any reason that we couldn't just decide one or the other and have a single context.

Reply:
Is there a reason why we wouldn't use Reach Router's location to get the state and querystring? (`useLocation().search`)?

Reply:
There is not. Either is fine. I didn't realize we relying so heavily on reach router when I did the relier work. We should swap this out. I'll file a follow up.

Reply:
File up followed: FXA-7048